cpp/CMakeLists.txt

```diff
@@ -170,6 +170,10 @@ static|shared (default shared)")
     "If off, 'quiet' flags will be passed to linting tools"
     OFF)
 
+  option(ARROW_GGDB_DEBUG
+    "Pass -ggdb flag to debug builds"
+    OFF)
```

Comment:
I don't know enough about the debug symbol flags to know what happens when `-ggdb` is not passed, do you have a reference?

Reply:
it should probably be just -g instead of -ggdb. -ggdb produces gdb-specifically for gdb (from [here, gcc but same difference](http://gcc.gnu.org/onlinedocs/gcc/Debugging-Options.html#Debugging-Options))

Reply:
Hm. So I would suggest having a flag about whether to add `-g` (with default being ON). We might need another option to customize the flag passed, but this could also be passed with ARROW_CXXFLAGS, too

Reply:
I want to use `-g3` instead of `-ggdb` or `-g`.
We can expand macro in GDB with `-g3`.

Does `-g3` work  with Emscripten?

Reply:
it does, for now i've just done -g and -ggdb